polygon/sync/sync.go

```diff
@@ -699,6 +706,30 @@ func (s *Sync) syncToTip(ctx context.Context) (syncToTipResult, error) {
 		return syncToTipResult{}, err
 	}
 
+	// if the bridge does not have all events yet we need to its
```

Comment:
This is already handled in `bridgeReplayInitialBlockIfNeeded`
I don't think we need this here unless there is a weird case I'm missing?
We end up having 2 places that take care of the same situation which doesn't feel right. We need to pick 1.

Reply:
bridgeReplayInitialBlockIfNeeded only syncs to the block with the latest block with an event in it.  This may be lower than the current block in snapshots for example, because not all blocks have events.  In any case it will only sync to the end of the last sprint it processed leaving a gap.

Both of these situations leave a gap as far as the bridge is concerned.  So the bridge will blow up on insert blocks as it detects a gap.

It feels a bit like an unhanded edge case to me.  I don't have a string opinion at to where the code here goes.  But something needs to detect and deal with the gap between sync & the bridge.